jcl/src/java.base/share/classes/jdk/internal/foreign/abi/UpcallLinker.java

```diff
@@ -129,12 +111,16 @@ public static UpcallStubFactory makeFactory(MethodType methodType, ABIDescriptor
 	 *
 	 * @param methodType the MethodType of the upcall method handle
 	 * @param descriptor the FunctionDescriptor of the upcall method handle
+	 * @param options the LinkerOptions indicating additional linking requirements to the linker
 	 * @return a factory instance that wraps up the upcall specific code
+	 * @throws InternalError as the upcalll specific code is not yet implemented
 	 */
-	public static UpcallStubFactory makeFactory(MethodType methodType, FunctionDescriptor descriptor) {
-		return (target, session) -> {
-			return UpcallLinker.make(target, methodType, descriptor, session);
-		};
+	@SuppressWarnings("nls")
+	public static UpcallStubFactory makeFactory(MethodType methodType, FunctionDescriptor descriptor, LinkerOptions options) {
+		// return (target, arena) -> {
+		//	return UpcallLinker.make(target, methodType, descriptor, arena, options);
+		// };
+		throw new InternalError("Upcall is not yet implemented"); //$NON-NLS-1$
```

Comment:
With `@SuppressWarnings("nls")` added above, the comment `//$NON-NLS-1$` is at best redundant, please remove `//$NON-NLS-1$` here and in similar situations (I don't think `javac` understands `//$NON-NLS-1$`, so `@SuppressWarnings("nls")` is preferable.)

Reply:
Agreed and updated everywhere against the suggestion above.

Reply:
It appears you did the opposite of what I suggested (to keep `SuppressWarnings("nls")` and remove redundant `NON-NLS` comments).